app/routers/categories.py

```diff
@@ -0,0 +1,80 @@
+from typing import Dict, List
+
+from fastapi import APIRouter, Depends, HTTPException, Request
+from pydantic import BaseModel
+from sqlalchemy.exc import IntegrityError, SQLAlchemyError
+from sqlalchemy.orm import Session
+from starlette import status
+from starlette.datastructures import ImmutableMultiDict
+
+from app.database.database import get_db
+from app.database.models import Category
+
+router = APIRouter(
+    prefix="/categories",
+    tags=["categories"],
+)
+
+
+class CategoryModel(BaseModel):
+    name: str
+    color: str
+    user_id: int
+
+
+# TODO(issue#29): get current user_id from session
+@router.get("/")
+def get_categories(request: Request,
+                   db_session: Session = Depends(get_db)) -> List[Category]:
+    if validate_request_params(request.query_params):
+        return get_user_categories(db_session, **request.query_params)
+    else:
+        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST,
+                            detail=f"Request {request.query_params} contains "
+                                   f"unallowed params.")
+
+
+# TODO(issue#29): get current user_id from session
+@router.post("/")
+async def set_category(category: CategoryModel,
+                       db_session: Session = Depends(get_db)) -> Dict:
```

Comment:
Please add more specification to the annotation regarding the values in the `Dict`